app/Response/Calendar.hs

```diff
@@ -12,12 +12,14 @@ import Data.Time.Calendar.OrdinalDate (fromMondayStartWeek, mondayStartWeek)
 import Database.CourseQueries (returnMeeting)
 import Database.Persist.Sqlite (entityKey, entityVal, runSqlite, selectList, (==.))
 import Database.Tables
-import Happstack.Server (Response, ServerPart, toResponse)
+import Happstack.Server (Response, ServerPart, toResponse, look)
 import Text.Read (readMaybe)
 
 -- | Returns an ICS file of events as requested by the user.
-calendarResponse :: String -> ServerPart Response
-calendarResponse = liftIO . getCalendar . T.pack
+calendarResponse :: ServerPart Response
+calendarResponse = do
+    courses <- look "courses"
```

Comment:
Now that everything's combined in this function, take this opportunity to simplify a bit here by using `lookText'`